server.go

```diff
@@ -202,20 +188,63 @@ func getRequestBody(input string) (*http.Request, error) {
 	inputs = append(inputs, data)
 	reqBody := RequestBody{inputs}
 	fmt.Print(reqBody)
-	//reqBodyBytes, err := json.Marshal(reqBody)
-	//if err != nil {
-		//return nil, err
-	//}
 
 	b := new(bytes.Buffer)
-  err := json.NewEncoder(b).Encode(reqBody)
+	err := json.NewEncoder(b).Encode(reqBody)
 	req, err := http.NewRequest("POST", clarifaiApi, b)
 	fmt.Print(req)
 	if err != nil {
 		return nil, err
 	}
 
-	req.Header.Add("Authorization", `Bearer ThzzqYyVARtJdLbTQDwfRpa1FOk8w6`)
+	req.Header.Add("Authorization", `Bearer ` + clarifai_token.Token)
 	req.Header.Add("Content-Type", `application/json`)
 	return req, nil
 }
+
+func getAccesToken() AccessToken {
+	CLARIFAI_ID := os.Getenv("CLARIFAI_CLIENT_ID")
+	CLARIFAI_SECRET_KEY := os.Getenv("CLARIFAI_SECRET_KEY")
+
+	responseData := authRequest(CLARIFAI_ID, CLARIFAI_SECRET_KEY)
+	var token AccessToken
+	rawIn := json.RawMessage(responseData)
+
+	bytes, err := rawIn.MarshalJSON()
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication response")
+		panic(err)
+	}
+
+	err = json.Unmarshal(bytes, &token)
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication json string")
+		panic(err)
+	}
+	fmt.Print("\nRESULTS: ")
+	fmt.Print(token.Token)
+	fmt.Print(token.ExpiresIn)
+
+	return token
+}
+
+func authRequest(CLIENT_ID string, SECRET_KEY string) []byte {
+
+	resp, err := http.PostForm(clarifyAuth, url.Values{"grant_type": {"client_credentials"},
+		"client_id": { CLIENT_ID },
+		"client_secret": { SECRET_KEY } })
+
+	fmt.Print("\nAuthentication Request constructed and sent")
+
+	if err != nil {
+		panic(err)
```

Comment:
@eladmica 
answering your questions:
Yah, ideally we shouldn't. But before it was hardcoded so this is kind of the first step to move away from that. API token expires in certain amount of minutes, that's why I made a struct that gets the token and it's expiration time. So, ideally, we should be requesting a new token once existing one expires, but i haven't implemented that logic yet, so for now it requests it with every API call (although it seems to return the same token while it's valid)